### ISSUE TITLE
[P2] Public ctm, ctm_2site, and ctm_split ignore CTMConfig.min_iter

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#976]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

## Finding

All three legacy array-based CTM entry points can declare convergence before the configured minimum sweep count. `CTMConfig.min_iter` explicitly promises a minimum before checking convergence, but none of these loops reads it. The two-site path is also used by `ipeps()` for simple-update energy measurement.

Verified at `4090f98be5c3efdce20a28663536f926171f857b`, CPU/JAX x64.

## Reproduction

Run with `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`:

```python
import jax.numpy as jnp
from tenax import CTMConfig, ctm, ctm_2site, ctm_split

A = jnp.ones((1, 1, 1, 1, 2))
for minimum in (2, 10, 30):
    cfg = CTMConfig(chi=1, max_iter=40, min_iter=minimum)
    print(minimum, "ctm", ctm(A, cfg, return_meta=True)[1])
    print(minimum, "ctm_2site", ctm_2site(A, A, cfg, return_meta=True)[2])
    print(minimum, "ctm_split", ctm_split(A, cfg, return_meta=True)[1])
```

Every call reports `converged=True, n_iter=2`, including `min_iter=30`. The residual is approximately `1.11e-16`.

Expected: perform at least the configured minimum before declaring convergence, while respecting `max_iter`. The product state makes this a cheap deterministic contract test; it is not a claim that its exact energy requires 30 sweeps. On transient spectra, the missing minimum removes the caller's requested safeguard against premature stopping.

## Root Cause

- [Configuration contract](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/ipeps_config.py#L86).
- [`ctm` loop](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/ipeps_ctm_convergence.py#L192-L206): the convergence predicate depends only on `diff < conv_tol`.
- [`ctm_2site` loop](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/ipeps_ctm_convergence.py#L350-L365): same omission.
- [`ctm_split` loop](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/ipeps_ctm_convergence.py#L508-L533): breaks on the tolerance test without checking `n_iter`.
- The shared Tensor-protocol loop already applies `min_iter`, so the same configuration behaves differently across engines.

## Suggested Fix and Coverage

Gate successful convergence on the total sweep count reaching `config.min_iter`. In `ctm`, include QR warmup in that total. Continue updating the previous spectrum during the minimum window.

Add parameterized tests for all three public entry points: `min_iter=30, max_iter=40`, a budget below the minimum, and ordinary early convergence once the minimum is met. Include a QR warmup boundary test.

Related but separate: #925 concerns incorrect QR sweep accounting in metadata. This omission reproduces with no QR warmup and changes when the algorithm stops, not just its reported count.
